packages/blockchain/src/index.ts

```diff
@@ -1135,6 +1166,70 @@ export default class Blockchain implements BlockchainInterface {
       return false
     }
   }
+
+  /**
+   * Returns the genesis {@link Block} for the blockchain.
+   * @param stateRoot When initializing the block, pass the genesis stateRoot.
+   *                  After the blockchain is initialized, this parameter is not used
+   *                  as the cached genesis block is returned.
+   */
+  genesisBlock(stateRoot?: Buffer): Block {
+    if (this._genesisBlock) {
+      return this._genesisBlock
+    }
+
+    if (!stateRoot) throw new Error('stateRoot required for genesis block creation')
+
+    const common = this._common.copy()
+    common.setHardforkByBlockNumber(0)
+
+    const { gasLimit, timestamp, difficulty, extraData, nonce, baseFeePerGas } = common.genesis()
+
+    const header: BlockData['header'] = {
+      number: 0,
+      gasLimit: BigInt(gasLimit),
+      timestamp: BigInt(timestamp ?? 0),
+      difficulty: BigInt(difficulty),
+      extraData: toBuffer(extraData),
+      nonce: toBuffer(nonce),
+      stateRoot,
+    }
+
+    if (baseFeePerGas !== undefined && common.gteHardfork(Hardfork.London)) {
+      header.baseFeePerGas = BigInt(baseFeePerGas)
+    }
+
+    return Block.fromBlockData({ header }, { common })
```

Comment:
ah yeah really good point actually, and we don't need to do any bigint or buffer conversions since fromBlockData will handle that, so to make this more future-proof I can do something like:

```
const header: BlockData['header'] = {
      ...common.genesis(),
      number: 0,
      stateRoot
}
```

I will add in a small follow up PR with a few other small fixes on my develop todo list.